File: 01p2.swift
var sum = 0;
var second = 0;
var ds: [Int] = [];
var seenFreq: [Int: Bool] = [:]
var secondFound = false;
while let dStr = readLine() {
  let d: Int = Int(dStr)!
  ds.append(d)
}
while !secondFound {
  for d in ds {
    sum += d
    if let f = seenFreq[sum] {
      second = sum
      secondFound = true
      break
    } else {
      seenFreq[sum] = true
    }
  }
}
print("\(second)")

File: python/01p2.py
#!/usr/bin/env python

import sys

sum = 0
second = sum
ds = []
seen_freq = {}
second_found = False
for d in sys.stdin: ds.append(int(d))
while not second_found:
    for d in ds:
        sum += d
        if sum in seen_freq:
            second = sum
            second_found = True
            break
        else:
            seen_freq[sum] = True
print second
